lib/src/extensions.dart

```diff
@@ -0,0 +1,6 @@
+import 'characters.dart';
+
+extension FromString on String {
+  /// Creates a [Characters] from the current String.
```

Comment:
Should be a getter, and therefore be documented as a property:
```dart
/// The [Characters] of this string.
Characters get characters => Characters(this);
```
(You even used it as a getter in the example above).

We could use both `asCharacters()` (adaption) and `toCharacters()` (conversion), but conceptually I think of a string as consisting of the characters, so this is simply a property, not a conversion or adaption of the string, it's its *content*.